xonsh/main.py

```diff
@@ -386,6 +372,20 @@ def func_sig_ttin_ttou(n, f):
             code = sys.stdin.read()
             run_code_with_cache(code, shell.execer, glb=shell.ctx, loc=None,
                                 mode='exec')
+        if args.force_interactive:
```

Comment:
we want xonsh to run the command or script before going into interactive,
modes are exclusive, we need new modes that correspond to 'interactive with script' and 'interactive with command'. With command/script argument present premain just sets mode accordingly so we can't check against mode for interactive. I was checking both against `args.force_interactive` and `args.mode` but it seemed redundant so i removed it.